src/pages/CollectionSingleViewPage.tsx

```diff
@@ -359,13 +402,149 @@ const TabPanel = ({ children, value, index, ...other }: TabPanelProps) => {
   );
 };
 
+type MutationTableTabContentProps = {
+  mutationType: string;
+  locationSelector: LocationSelector;
+  variants: { query: VariantSelector; name: string; description: string }[];
+  dateRangeSelector: DateRangeSelector;
+  mode: 'Single' | 'CompareToBaseline';
+  baselineVariant?: VariantSelector;
+};
+
+const MutationTableTabContent = ({
```

Comment:
Can we maybe cut this component in smaller pieces? I'd propose the following structure (in pseudocode):

```
function MutationTableTabContent (props) {
  return mode === 'Single' ? <SingleMutationTableTabContent /> : <BaselineMutationTableTabContent  />
} 

function SingleMutationTableTabContent() {
  data = useMutationsData(selectors, variants, ...)
  return <SingleMutationsTable data={data} />
}

function BaselineMutationTableTabContent() {
  data = useBaselineMutationsData(selectors, variants, ...)
  return <BaselineMutationsTable data={data}
}

function useBaselineMutationsData(...) {
  mutationsData = useMutationsData(selectors, variants, ...)

  // other useQuery, useMemo etc. for baseline data only

  return { mutationsData , baselineData }
}

function useMutationsData(...) {
  // useQuery, useMemo etc.
  return mutationsData 
}
```

Maybe even in a separate file? The current file grew to 800 lines which is quite a lot.

Reply:
The downside is that we lose caching that way, because the self-written useQuery doesn't have a cache. However the browser seems to have some basic caching there, so we might save some requests.

Using `react-query` would really help here, because it has a proper built-in cache.